src/teacher.py

```diff
@@ -75,6 +80,81 @@ def _answer_Demonstration_query(
                                           _train_task_idx=query.train_task_idx)
         return DemonstrationResponse(query, teacher_traj)
 
+    def _answer_StateBasedDemonstration_query(
+            self, state: State,
+            query: StateBasedDemonstrationQuery) -> DemonstrationResponse:
+        # The query is asking for a demonstration from the current state to
+        # the goal state. Planning to a low-level state is hard. This
+        # implementation currently assumes that only one option is required
+        # to get from the state to the goal state. Furthermore, the option
+        # is identified in an environment-specific way. This is all a stand-in
+        # for a human teacher. Note that although these trajectories are good,
+        # they are not demonstrations per se, because they do not reach task
```

Comment:
huh? how are they not demonstrations? then this class is named incorrectly (but i do think they are demonstrations. why do demonstrations necessarily have to be associated with a task?)